src/chains/ethereum/transaction/tests/index.test.ts

```diff
@@ -408,7 +408,7 @@ describe("@ganache/ethereum-transaction", async () => {
       tx.signAndHash(privKeyBuf);
       assert.strictEqual(
         tx.hash.toString(),
-        "0x078395f79508111c9061f9983d387c8b7bfed990dfa098497aa4d34b0e47b265"
+        "0x1d37ff36e8af71fca608ad0f92b52e741b381663ebfe02d0dd1117a4462c206f"
```

Comment:
Ah, probably should have added a comment on that - the hashes have changed because we now respect bytelength on toBuffer. There's some fixed length buffers fed into the hash function, so previously they were trimmed.

Reply:
Oooohhh. This might be a mistake. IIRC Buffers are supposed to be compact when hashing (leading 0s removed). I might be wrong, but we need to double check!

Reply:
I looked further into this, the "offending" statement is here: https://github.com/trufflesuite/ganache/blob/d1b9bbda7bcc3ca12973c93313d78365e8b82c98/src/chains/ethereum/transaction/src/access-lists.ts#L30

If I remove the bytelength argument, making it (previously this was ignored anyways):
`const addressBuffer = Data.toBuffer(item.address);`

The hash is `0x078395f79508111c9061f9983d387c8b7bfed990dfa098497aa4d34b0e47b265` which is the previously expected value.

v, r, s all use `Quantity` type, so will explicitly strip leading 0 bytes however.

Reply:
> The [EIP-2718](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2718.md) `TransactionPayload` for this transaction is `rlp([chainId, nonce, gasPrice, gasLimit, to, value, data, accessList, signatureYParity, signatureR, signatureS])`.
> ...
> For the transaction to be valid, `accessList` must be of type `[[{20 bytes}, [{32 bytes}...]]...]`, where `...` means "zero or more of the thing to the left".
Per https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2930.md

Looks like we _do_ need to ensure a correct `byteLength`, but the `byteLength` should be `20` (so we can use `Address.toBuffer` here?).

Reply:
I've added a comment to this effect - as well as enforcing that the accesslist storagekey part is 32 bytes